shell/platform/android/io/flutter/plugin/common/BinaryMessenger.java

```diff
@@ -116,6 +116,22 @@ default void setMessageHandler(
     setMessageHandler(channel, handler);
   }
 
+  /**
+   * Enables the ability to queue messages received from Dart.
+   *
+   * <p>This is useful when there are pending channel handler registrations. For example, Dart may
+   * be initialized concurrently, and prior to the registration of the channel handlers. This
+   * implies that Dart may start sending messages while plugins are being registered.
+   */
+  void enableBufferingIncomingMessages();
```

Comment:
There going to be another PR that will enable and disable the buffering during the timeframe of plugin registration I imagine?

Reply:
that's right